src/main/java/brave/features/opentracing/BraveTracer.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.features.opentracing;
+
+import brave.internal.Nullable;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.TracerAdapter;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BraveTracer implements Tracer {
+
+    static final List<String> PROPAGATION_KEYS = Propagation.B3_STRING.keys();
+    static final TraceContext.Injector<TextMap> INJECTOR = Propagation.B3_STRING.injector(TextMap::put);
+    static final TraceContext.Extractor<TextMapView> EXTRACTOR = Propagation.B3_STRING.extractor(TextMapView::get);
+
+    private brave.Tracer brave4;
+    private Brave brave3; // Not sure if I need this or not. Might want to attach to the local tracer context
```

Comment:
I think the adapter pattern is sufficient rather than make a permanent dependency on brave-core